codegen/src/lut.rs

```diff
@@ -0,0 +1,470 @@
+        is_linear_as_until: Option<(f64, f64)>,
+        gamma: f64,
+    ) -> Self {
+        let (linear_scale, alpha, beta) =
+            if let Some((linear_scale, linear_end)) = is_linear_as_until {
+                (
+                    Some(linear_scale),
+                    (linear_scale * linear_end - 1.0) / (linear_end.powf(gamma.recip()) - 1.0),
+                    linear_end,
+                )
+            } else {
+                (None, 1.0, 0.0)
+            };
+        Self {
+            fn_type: format_ident!("{fn_type}"),
+            fn_type_uppercase: fn_type_uppercase.to_owned(),
+            into_linear: Box::new(move |encoded| match linear_scale {
+                Some(scale) if encoded <= scale * beta => encoded / scale,
+                _ => ((encoded + alpha - 1.0) / alpha).powf(gamma),
+            }),
+            linear_scale,
+            alpha,
+            beta,
+            gamma,
+        }
+    }
+}
+
+impl LutEntryU16 {
+    fn new(
+        fn_type: &str,
+        fn_type_uppercase: &str,
+        is_linear_as_until: Option<(f64, f64)>,
+        gamma: f64,
+    ) -> Self {
+        let (linear_scale, alpha, beta) =
+            if let Some((linear_scale, linear_end)) = is_linear_as_until {
+                (
+                    Some(linear_scale),
+                    (linear_scale * linear_end - 1.0) / (linear_end.powf(gamma.recip()) - 1.0),
+                    linear_end,
+                )
+            } else {
+                (None, 1.0, 0.0)
+            };
+        Self {
+            fn_type: format_ident!("{fn_type}"),
+            fn_type_uppercase: fn_type_uppercase.to_owned(),
+            into_linear: Box::new(move |encoded| match linear_scale {
+                Some(scale) if encoded <= scale * beta => encoded / scale,
+                _ => ((encoded + alpha - 1.0) / alpha).powf(gamma),
+            }),
+            linear_scale,
+            alpha,
+            beta,
+            gamma,
+        }
+    }
+}
+
+fn build_u8_to_f64_lut(entries: &[LutEntryU8]) -> TokenStream {
+    let tables = entries.iter().map(
+        |LutEntryU8 {
+             fn_type,
+             fn_type_uppercase,
+             into_linear,
+             ..
+         }| {
+            let table = (0..=u8::MAX).map(|i| into_linear((i as f64) / 255.0));
+            let table_ident = format_ident!("{fn_type_uppercase}_U8_TO_F64");
+            quote! {
+                const #table_ident: [f64; 256] = [
+                    #(#table),*
+                ];
+
+                impl IntoLinear<f64, u8> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u8) -> f64 {
+                        #table_ident[encoded as usize]
+                    }
+                }
+
+                impl IntoLinear<f32, u8> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u8) -> f32 {
+                        #table_ident[encoded as usize] as f32
```

Comment:
Let's try adding an `f32` table to be used here, and see if that has any effect on the benchmark that got a slow-down.